### PR TITLE
Set min transformers version when importing

### LINKED ISSUE #1945: Check for `transformers` is not precise enough
## 🐛 Bug

Hello! I've got a MacBook Pro, 13-inch, 2020, Four Thunderbolt 3 ports, with an Intel CPU.
I've found that with the latest version of Anaconda distribution (`Anaconda3-2023.07-1-MacOSX-x86_64.pkg`, downloaded from anaconda.com about an hour ago), the base environment comes with a surprisingly out-of-date version of `transformers`:

```
(base) steve@Stevens-MacBook-Pro ~ % pip list | grep transformers
transformers                  2.1.1
```

This version does not have the module structure that `torchmetrics` version `1.0.1` assumes, and a simple import like `import torchmetrics` will raise an exception.

### To Reproduce

* Download and install Anaconda for Mac Intel from anaconda.com
* `pip install torchmetrics`
* Open python
* `import torchmetrics`
* An `ImportError` is raised

I get the following:

```
(base) steve@Stevens-MacBook-Pro ~ % pip install torchmetrics
Collecting torchmetrics
  Using cached torchmetrics-1.0.1-py3-none-any.whl (729 kB)
Requirement already satisfied: numpy>1.20.0 in ./anaconda3/lib/python3.11/site-packages (from torchmetrics) (1.24.3)
Collecting torch>=1.8.1 (from torchmetrics)
  Using cached torch-2.0.1-cp311-none-macosx_10_9_x86_64.whl (143.1 MB)
Requirement already satisfied: packaging in ./anaconda3/lib/python3.11/site-packages (from torchmetrics) (23.0)
Collecting lightning-utilities>=0.7.0 (from torchmetrics)
  Using cached lightning_utilities-0.9.0-py3-none-any.whl (23 kB)
Requirement already satisfied: typing-extensions in ./anaconda3/lib/python3.11/site-packages (from lightning-utilities>=0.7.0->torchmetrics) (4.6.3)
Requirement already satisfied: filelock in ./anaconda3/lib/python3.11/site-packages (from torch>=1.8.1->torchmetrics) (3.9.0)
Requirement already satisfied: sympy in ./anaconda3/lib/python3.11/site-packages (from torch>=1.8.1->torchmetrics) (1.11.1)
Requirement already satisfied: networkx in ./anaconda3/lib/python3.11/site-packages (from torch>=1.8.1->torchmetrics) (2.8.4)
Requirement already satisfied: jinja2 in ./anaconda3/lib/python3.11/site-packages (from torch>=1.8.1->torchmetrics) (3.1.2)
Requirement already satisfied: MarkupSafe>=2.0 in ./anaconda3/lib/python3.11/site-packages (from jinja2->torch>=1.8.1->torchmetrics) (2.1.1)
Requirement already satisfied: mpmath>=0.19 in ./anaconda3/lib/python3.11/site-packages (from sympy->torch>=1.8.1->torchmetrics) (1.2.1)
Installing collected packages: lightning-utilities, torch, torchmetrics
Successfully installed lightning-utilities-0.9.0 torch-2.0.1 torchmetrics-1.0.1
(base) steve@Stevens-MacBook-Pro ~ % python
Python 3.11.3 (main, Apr 19 2023, 18:51:09) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import torchmetrics
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/steve/anaconda3/lib/python3.11/site-packages/torchmetrics/__init__.py", line 14, in <module>
    from torchmetrics import functional  # noqa: E402
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/steve/anaconda3/lib/python3.11/site-packages/torchmetrics/functional/__init__.py", line 121, in <module>
    from torchmetrics.functional.text._deprecated import _bleu_score as bleu_score
  File "/Users/steve/anaconda3/lib/python3.11/site-packages/torchmetrics/functional/text/__init__.py", line 31, in <module>
    from torchmetrics.functional.text.bert import bert_score
  File "/Users/steve/anaconda3/lib/python3.11/site-packages/torchmetrics/functional/text/bert.py", line 25, in <module>
    from torchmetrics.functional.text.helper_embedding_metric import (
  File "/Users/steve/anaconda3/lib/python3.11/site-packages/torchmetrics/functional/text/helper_embedding_metric.py", line 27, in <module>
    from transformers import AutoModelForMaskedLM, AutoTokenizer, PreTrainedModel, PreTrainedTokenizerBase
ImportError: cannot import name 'AutoModelForMaskedLM' from 'transformers' (/Users/steve/anaconda3/lib/python3.11/site-packages/transformers/__init__.py)
>>> 
```

### Expected behavior

The import should not raise an exception.

This can be achieved by upgrading `transformers`:
```
(base) steve@Stevens-MacBook-Pro ~ % pip install --upgrade transformers
Requirement already satisfied: transformers in ./anaconda3/lib/python3.11/site-packages (2.1.1)
Collecting transformers
  Using cached transformers-4.31.0-py3-none-any.whl (7.4 MB)
Requirement already satisfied: filelock in ./anaconda3/lib/python3.11/site-packages (from transformers) (3.9.0)
Collecting huggingface-hub<1.0,>=0.14.1 (from transformers)
  Using cached huggingface_hub-0.16.4-py3-none-any.whl (268 kB)
Requirement already satisfied: numpy>=1.17 in ./anaconda3/lib/python3.11/site-packages (from transformers) (1.24.3)
Requirement already satisfied: packaging>=20.0 in ./anaconda3/lib/python3.11/site-packages (from transformers) (23.0)
Requirement already satisfied: pyyaml>=5.1 in ./anaconda3/lib/python3.11/site-packages (from transformers) (6.0)
Requirement already satisfied: regex!=2019.12.17 in ./anaconda3/lib/python3.11/site-packages (from transformers) (2022.7.9)
Requirement already satisfied: requests in ./anaconda3/lib/python3.11/site-packages (from transformers) (2.29.0)
Collecting tokenizers!=0.11.3,<0.14,>=0.11.1 (from transformers)
  Using cached tokenizers-0.13.3-cp311-cp311-macosx_10_11_universal2.whl (4.0 MB)
Collecting safetensors>=0.3.1 (from transformers)
  Using cached safetensors-0.3.1-cp311-cp311-macosx_10_11_universal2.whl (400 kB)
Requirement already satisfied: tqdm>=4.27 in ./anaconda3/lib/python3.11/site-packages (from transformers) (4.65.0)
Requirement already satisfied: fsspec in ./anaconda3/lib/python3.11/site-packages (from huggingface-hub<1.0,>=0.14.1->transformers) (2023.3.0)
Requirement already satisfied: typing-extensions>=3.7.4.3 in ./anaconda3/lib/python3.11/site-packages (from huggingface-hub<1.0,>=0.14.1->transformers) (4.6.3)
Requirement already satisfied: charset-normalizer<4,>=2 in ./anaconda3/lib/python3.11/site-packages (from requests->transformers) (2.0.4)
Requirement already satisfied: idna<4,>=2.5 in ./anaconda3/lib/python3.11/site-packages (from requests->transformers) (3.4)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./anaconda3/lib/python3.11/site-packages (from requests->transformers) (1.26.16)
Requirement already satisfied: certifi>=2017.4.17 in ./anaconda3/lib/python3.11/site-packages (from requests->transformers) (2023.5.7)
Installing collected packages: tokenizers, safetensors, huggingface-hub, transformers
  Attempting uninstall: transformers
    Found existing installation: transformers 2.1.1
    Uninstalling transformers-2.1.1:
      Successfully uninstalled transformers-2.1.1
Successfully installed huggingface-hub-0.16.4 safetensors-0.3.1 tokenizers-0.13.3 transformers-4.31.0
(base) steve@Stevens-MacBook-Pro ~ % python
Python 3.11.3 (main, Apr 19 2023, 18:51:09) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import torchmetrics
>>> 
```

### Environment

- TorchMetrics version: 1.0.1, via `pip`.
- Python version: Python 3.11.3 (main, Apr 19 2023, 18:51:09) [Clang 14.0.6 ] on darwin
- PyTorch version: 2.0.1
- OS: MacBook Pro, 13-inch, 2020, Four Thunderbolt 3 ports, 2 GHz Quad-Core Intel Core i5, macOS 13.4.1 (c) 

### Additional context

N/A

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #1945
Changing lines from
```python
if _TRANSFORMERS_AVAILABLE:
    from transformers import ...
```
to
```python
if _TRANSFORMERS_GREATER_EQUAL_XX:
    from transformers import ...
```

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--1947.org.readthedocs.build/en/1947/

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

File: src/torchmetrics/functional/text/helper_embedding_metric.py [hunk 1]
```diff
@@ -21,9 +21,9 @@
 from torch.utils.data import DataLoader, Dataset
 
 from torchmetrics.utilities.data import _cumsum
-from torchmetrics.utilities.imports import _TQDM_AVAILABLE, _TRANSFORMERS_AVAILABLE
+from torchmetrics.utilities.imports import _TQDM_AVAILABLE, _TRANSFORMERS_GREATER_EQUAL_4_10
```
Comment (stancld) [inline src/torchmetrics/functional/text/helper_embedding_metric.py:24]:
```suggestion
from torchmetrics.utilities.imports import _TQDM_AVAILABLE, _TRANSFORMERS_GREATER_EQUAL_4_4
```

File: src/torchmetrics/functional/text/helper_embedding_metric.py [hunk 2]
```diff
@@ -21,9 +21,9 @@
...
 
-if _TRANSFORMERS_AVAILABLE:
+if _TRANSFORMERS_GREATER_EQUAL_4_10:
```
Comment (stancld) [inline src/torchmetrics/functional/text/helper_embedding_metric.py:26]:
```suggestion
if _TRANSFORMERS_GREATER_EQUAL_4_4:
```

File: tests/unittests/multimodal/test_clip_score.py [hunk 3]
```diff
@@ -55,7 +55,7 @@ def _compare_fn(preds, target, model_name_or_path):
 
 @pytest.mark.parametrize("model_name_or_path", ["openai/clip-vit-base-patch32"])
 @pytest.mark.parametrize("inputs", [_random_input])
-@pytest.mark.skipif(not _TRANSFORMERS_AVAILABLE, reason="test requires bert_score")
+@pytest.mark.skipif(not _TRANSFORMERS_GREATER_EQUAL_4_10, reason="test requires bert_score")
```
Comment (stancld) [inline tests/unittests/multimodal/test_clip_score.py:58]:
```suggestion
@pytest.mark.skipif(not _TRANSFORMERS_GREATER_EQUAL_4_10, reason="test requires transformers>=4.10")
```

File: tests/unittests/text/test_bertscore.py [hunk 4]
```diff
@@ -42,7 +42,7 @@
 os.environ["TOKENIZERS_PARALLELISM"] = "false"
 
 
-@pytest.mark.skipif(not _TRANSFORMERS_AVAILABLE, reason="test requires transformers")
+@pytest.mark.skipif(not _TRANSFORMERS_GREATER_EQUAL_4_4, reason="test requires transformers")
```
Comment (stancld) [inline tests/unittests/text/test_bertscore.py:45]:
```suggestion
@pytest.mark.skipif(not _TRANSFORMERS_GREATER_EQUAL_4_4, reason="test requires transformers>4.4")
```

File: tests/unittests/text/test_bertscore.py [hunk 5]
```diff
@@ -86,7 +86,7 @@ def _reference_bert_score(
     ["preds", "targets"],
     [(_inputs_single_reference.preds, _inputs_single_reference.targets)],
 )
-@pytest.mark.skipif(not _TRANSFORMERS_AVAILABLE, reason="test requires transformers")
+@pytest.mark.skipif(not _TRANSFORMERS_GREATER_EQUAL_4_4, reason="test requires transformers")
```
Comment (stancld) [inline tests/unittests/text/test_bertscore.py:89]:
```suggestion
@pytest.mark.skipif(not _TRANSFORMERS_GREATER_EQUAL_4_4, reason="test requires transformers>4.4")
```

File: tests/unittests/text/test_infolm.py [hunk 6]
```diff
@@ -99,7 +99,7 @@ def reference_infolm_score(preds, target, model_name, information_measure, idf,
     ["preds", "targets"],
     [(_inputs_single_reference.preds, _inputs_single_reference.targets)],
 )
-@pytest.mark.skipif(not _TRANSFORMERS_AVAILABLE, reason="test requires transformers")
+@pytest.mark.skipif(not _TRANSFORMERS_GREATER_EQUAL_4_4, reason="test requires transformers")
```
Comment (stancld) [inline tests/unittests/text/test_infolm.py:102]:
```suggestion
@pytest.mark.skipif(not _TRANSFORMERS_GREATER_EQUAL_4_4, reason="test requires transformers>=4.4")
```

**Review by stancld — COMMENTED** [review]
Maybe, we can also add `transformers` dependency to `requirements/text.txt` instead of `test_text.txt`. It's not a too heavy one

**Comment by SkafteNicki** [comment]
> Maybe, we can also add `transformers` dependency to `requirements/text.txt` instead of `test_text.txt`. It's not a too heavy one

Agree, I have moved the dependency :)

**Review by stancld — APPROVED** [review]

**Pull request merged**